app/src/androidTest/java/org/mozilla/focus/activity/screenshots/BrowserScreenScreenshots.java

```diff
@@ -0,0 +1,207 @@
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.hasFocus;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static junit.framework.Assert.assertTrue;
+import static org.hamcrest.Matchers.containsString;
+
+@RunWith(AndroidJUnit4.class)
+public class BrowserScreenScreenshots extends ScreenshotTest {
+    @Rule
+    public ActivityTestRule<MainActivity> mActivityTestRule = new MainActivityFirstrunTestRule(false);
+
+    @ClassRule
+    public static final LocaleTestRule localeTestRule = new LocaleTestRule();
+
+    private MockWebServer webServer;
+
+    @Before
+    public void setUpWebServer() throws IOException {
+        webServer = new MockWebServer();
+
+        // Test page
+        webServer.enqueue(new MockResponse()
+                .setBody(TestHelper.readTestAsset("image_test.html")));
+        webServer.enqueue(new MockResponse()
+                .setBody(TestHelper.readTestAsset("rabbit.jpg")));
+        webServer.enqueue(new MockResponse()
+                .setBody(TestHelper.readTestAsset("download.jpg")));
+        // Download
+        webServer.enqueue(new MockResponse()
+                .setBody(TestHelper.readTestAsset("download.jpg")));
+    }
+
+    @After
+    public void tearDownWebServer() {
+        try {
+            webServer.close();
+            webServer.shutdown();
+        } catch (IOException e) {
+            throw new AssertionError("Could not stop web server", e);
+        }
+    }
+
+    @Test
+    public void takeScreenshotsOfBrowsingScreen() throws Exception {
+        takeScreenshotsOfBrowsingView();
+        takeScreenshotsOfMenu();
+        takeScreenshotsOfOpenWithAndShare();
+        takeAddToHomeScreenScreenshot();
+        takeScreenshotOfTabsTrayAndErase();
+    }
+
+    private void takeScreenshotsOfBrowsingView() {
+        assertTrue(TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime));
+        Screengrab.screenshot("LocationBarEmptyState");
+
+        /* Autocomplete View */
+        onView(withId(R.id.url_edit))
+                .check(matches(isDisplayed()))
+                .check(matches(hasFocus()))
+                .perform(click(), replaceText("mozilla"));
+
+        assertTrue(TestHelper.hint.waitForExists(waitingTime));
+
+        Screengrab.screenshot("SearchFor");
+
+        onView(withId(R.id.url_edit))
+                .check(matches(isDisplayed()))
+                .check(matches(hasFocus()))
+                .perform(click(), replaceText(webServer.url("/").toString()), pressImeActionButton());
+
+        device.findObject(new UiSelector()
+                .resourceId("org.mozilla.focus.debug:id/webview")
+                .enabled(true))
+                .waitForExists(waitingTime);
+
```

Comment:
let's also add check of progressBar to disappear. Sometimes webview appears, but it's still loading.  (esp. on slower sims)

public static UiObject progressBar = mDevice.findObject(new UiSelector()
            .resourceId("org.mozilla.focus.debug:id/progress")
            .enabled(true));